samwise.js

```diff
@@ -1,77 +1,81 @@
 /**
```

Comment:
SCRIPT-This logic might be clearer if you used a switch statement instead of multiple if-else statements.